sklearn/decomposition/pca.py

```diff
@@ -68,6 +69,9 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
         v = np.sum(spectrum[rank:]) / (n_features - rank)
         pv = -np.log(v) * n_samples * (n_features - rank) / 2.
 
+    if sys.float_info.epsilon > abs(v):
```

Comment:
Is it safe to use this, should we use [`np.finfo`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.finfo.html) instead, or should we just use a `tol` parameter or maybe a `rcond` as in [`scipy.pinv`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.pinv.html)?